### PR TITLE
chore(libsinsp): avoid useless allocation

### PR DESCRIPTION
**Pull request by therealbobo**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

This avoids the allocation of a temporary object: to do that `emplace_back` needs the arguments of the constructor underlying object.

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
> BM_sinsp_split_mean                                            -0.0706         -0.0706           151           140           151           140
BM_sinsp_split_median                                          -0.0709         -0.0709           151           140           151           140

Confirmed by benchmarks :)

**Review by FedeDP — DISMISSED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

File: userspace/libsinsp/test/async_key_value_source.ut.cpp [hunk 1]
```diff
@@ -358,15 +358,15 @@ TEST(async_key_value_source_test, prune_old_metadata) {
 	ASSERT_FALSE(source.lookup(key1, response));
 
 	// Wait long enough for the old entry to require pruning
-	std::this_thread::sleep_for(std::chrono::milliseconds(2 * TTL_MS));
+	usleep(1000 * 2 * TTL_MS);
```
[the remark this answers is not in the document]
Comment (therealbobo) [reply userspace/libsinsp/test/async_key_value_source.ut.cpp:361]: I was really out of ideas 😓 and I'm surprise as much as you. The problem should be related to the lifetime of the object and that was the only function that was related 😅

**Review by jasondellaluce — APPROVED** [review]
lol

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**